go/keyset/manager.go

```diff
@@ -46,16 +47,41 @@ func NewManagerFromHandle(kh *Handle) *Manager {
 
 // Rotate generates a fresh key using the given key template and
 // sets the new key as the primary key.
+// Deprecated: please use Add instead. Rotate adds a new key and immediately promotes it to primary.
+// However, when you do keyset rotation, you almost never want to make the new key primary,
+// because old binaries don't know the new key yet.
 func (km *Manager) Rotate(kt *tinkpb.KeyTemplate) error {
```

Comment:
Agreed, I figured adding the docstring here was best option since it's already part of public API

Reply:
I didn't think it was best to remove, it's part of the public API right now, so it would be a breaking change. The [Java](https://github.com/google/tink/blob/master/java_src/src/main/java/com/google/crypto/tink/KeysetManager.java#L72) and [C++ ](https://github.com/google/tink/blob/dca70f855d483d4350e222e2754db2f9e63c9ecb/cc/keyset_manager.h#L60)APIs currently expose this too (I mostly copied the Java comment). Could we keep this and then `Rotate` would be removed later from all APIs or perhaps all of these keyset APIs will be changed? I didn't think we'd want to break current implementation.

The migration isn't too bad but it's not a one-liner and I think requires more thought from clients, e.g.
```go
// original
err := manager.Rotate(template)
```
```go
// new
id, err := manager.Add(template)
// handle error and propagate new key material 😀
err = manager.SetPrimary(id)
```

What do you think?

Reply:
Ah you're right. I thought Rotate is a new method. Let's keep it then.